packages/run-openmrs/src/utils/index.ts

```
@@ -0,0 +1,73 @@
+import { resolve } from "path";
+import { existsSync, readFileSync } from "fs";
+
+export interface WebpackOptions {
+  backend?: string;
+  importmap?: ImportmapDeclaration;
+}
+
+export interface ImportmapDeclaration {
+  type: "inline" | "url";
+  value: string;
+}
+
+export function setEnvVariables(envVariables: Record<string, any>) {
+  Object.keys(envVariables).forEach((key) => {
+    process.env[key] = envVariables[key];
+  });
+}
+
+export function loadConfig(options: WebpackOptions = {}) {
+  const variables: Record<string, any> = {};
+
+  if (typeof options.backend === "string") {
+    variables.OMRS_PROXY_TARGET = options.backend;
+  }
+
+  if (typeof options.importmap === "object") {
+    switch (options.importmap.type) {
+      case "inline":
+        variables.OMRS_ESM_IMPORTMAP = options.importmap.value;
+        break;
+      case "url":
+        variables.OMRS_ESM_IMPORTMAP_URL = options.importmap.value;
+        break;
+    }
+  }
+
+  setEnvVariables(variables);
+
+  return require("@openmrs/esm-app-shell/webpack.config.js");
+}
+
+export function checkImportmapJson(value: string) {
+  try {
+    const content = JSON.parse(value);
+    return typeof content === "object" && typeof content.imports === "object";
+  } catch {
+    return false;
+  }
+}
+
+export function getImportmap(value: string): ImportmapDeclaration {
+  if (!/https?:\/\//.test(value)) {
+    const path = resolve(process.cwd(), value);
+
+    if (existsSync(path)) {
+      return {
+        type: "inline",
+        value: readFileSync(path, "utf8"),
+      };
+    } else if (checkImportmapJson(value)) {
```

Instead of letting this function return as if its a url if this check fails, can we throw an error with a helpful error message?

---

The check may fail if the provided string is like "my-import-map.json", which may be still valid at runtime. This is a valid (relative) URL.

The case where we may warn is if we know for sure that the name will be definitely an invalid URL or if the `readFileSync` fails the `checkImportmapJson`.

But for now I'd say lets leave it. If its really invalid it will be seen when running the shell anyway.

---

Added another check!